### ISSUE TITLE
Add `--v1-compatible` flag to more commands

### ISSUE THREAD

**Issue by johanfylling** [open-policy-agent/opa#6463]
When applied to commands other than `fmt` and `check`, the `--rego-v1` flag should enforce that _all_ supplied modules are compatible with OPA v1.0, but not require the `rego.v1` or `future.keywords` imports. 
Essentially, using the `--rego-v1` flag should make OPA parse, compile, and evaluate policies as if 1.0.

E.g. this is a valid module in OPA 1.0:

```rego
package example

# Notice: no imports

allow if {
  count(violations) == 0
}

violations contains msg if {
  ...
}
```

Commands:

* bench
* build
* capabilities
* check: already implemented
* completion: N/A
* deps
* eval
* exec
* fmt: already implemented
* help: N/A
* inspect
* parse
* run
* sign: N/A
* test
* version: N/A

**Comment by johanfylling** [comment]
It has been decided that we should use the `--v1-compatible` flag instead of `--rego-v1`.

**Comment by frek818** [comment]
@johanfylling, I notice that this issue is closed but the PR (#6520) is still open. Do you know if this feature will make into opa soon.

**Comment by johanfylling** [comment]
The scope of this issue was redefined to the `build`, `check`, `fmt`, `eval`, and `test` commands. I've updated the description to reflect this. 

PR #6521 implements #6520; which adds `--v1-compatible` support to the remaining applicable commands: `bench`, `deps`, `exec`, `inspect`, `run`, and `parse`. This is expected to be included in the next release (`0.61.0`).

I hope this clarifies things. Thanks for making us aware of the issue 😃.
